packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -383,6 +387,36 @@ Future<XcodeBuildResult> buildXcodeProject({
   );
   globals.flutterUsage.sendTiming(xcodeBuildActionToString(buildAction), 'xcode-ios', Duration(milliseconds: sw.elapsedMilliseconds));
 
+  if (tempDir.existsSync()) {
+    // Display additional warning and error message from xcresult bundle.
+    final Directory resultBundle = tempDir.childDirectory(_kResultBundlePath);
+    if (!resultBundle.existsSync()) {
+      globals.printTrace('The xcresult bundle are not generated. Displaying xcresult is disabled.');
+    } else {
+      final XCResultGenerator xcResultGenerator = XCResultGenerator(resultPath: resultBundle.absolute.path, xcode: globals.xcode!, processUtils: globals.processUtils);
+      final XCResult xcResult = await xcResultGenerator.generate();
+      if (xcResult.parseSuccess) {
+        if (xcResult.issues.isEmpty) {
+          globals.printStatus('Xcode build has no issues!');
+        }
+        for (final XCResultIssue issue in xcResult.issues) {
+          switch (issue.type) {
+            case XCResultIssueType.error:
+              globals.printError('Xcode build issue (${issue.subType}): ${issue.message}');
+              break;
+            case XCResultIssueType.warning:
+              globals.printWarning('Xcode build issue (${issue.subType}): ${issue.message}');
+              break;
```

Comment:
done

Reply:
Can you add examples for the full `flutter build ios` and `flutter build ios --verbose` output of what this kind of error will look like?  I can't picture how it will look in the context of all the Xcode logging and build failure text.

I would know more when I see the full output, but this seems too wordy: 
```
Xcode build issue (Swift Compiler Error): Expected dictionary value type
```

How about:
```
Swift Compiler Error (Xcode): Expected dictionary value type
```

1. Is this message going to be helpful without the class name and line number?  Unfortunately this means parsing the `url` result anchor (which looks like a query, oddly, thanks Xcode...) `file:///path/to/ios/Runner/AppDelegate.swift#CharacterRangeLen=0&EndingColumnNumber=8&EndingLineNumber=11&StartingColumnNumber=8&StartingLineNumber=11`
2. There are likely quite a few error messages we should not display, like `Command PhaseScriptExecution failed with a nonzero exit code` (would happen every time there's a dart compilation issue).  Same for the warning that happens when a plugin supports a lower iOS version than the app.  Let's get the obvious ones, and then we can add more in future PRs as we discover them.

Reply:
1. 
I think the compiler error usually display at the end of the build, so it looks like:
```
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        759ms
Xcode build done.                                            5.5s
Xcode build issue (Swift Compiler Error): Expected dictionary value type
Xcode build issue (Swift Compiler Error): Method does not override any method from its superclass
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/38/ypkvsh7n1d94tv_njrw0nwrh00l9tg/T/flutter_tools.CtyCRE/flutter_ios_build_temp_dire1EhUm/temporary_xcresult_bundle

    /Users/ychris/tmp/xcresult/xcresult_parsing/ios/Runner/AppDelegate.swift:8:83: error: expected dictionary value type
        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: nil]?
                                                                                      ^
    /Users/ychris/tmp/xcresult/xcresult_parsing/ios/Runner/AppDelegate.swift:6:17: error: method does not override any method from its superclass
      override func application(
      ~~~~~~~~      ^
    remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in parallel

    Result bundle written to path:
    	/var/folders/38/ypkvsh7n1d94tv_njrw0nwrh00l9tg/T/flutter_tools.CtyCRE/flutter_ios_build_temp_dire1EhUm/temporary_xcresult_bundle


════════════════════════════════════════════════════════════════════════════════
Building a deployable iOS app requires a selected Development Team with a 
Provisioning Profile. Please ensure that a Development Team is selected by:
  1- Open the Flutter project's Xcode target with
       open ios/Runner.xcworkspace
  2- Select the 'Runner' project in the navigator then the 'Runner' target
     in the project settings
  3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team. 
     You may need to:
         - Log in with your Apple ID in Xcode first
         - Ensure you have a valid unique Bundle ID
         - Register your device with your Apple Developer Account
         - Let Xcode automatically provision a profile for your app
  4- Build or run your project again

For more information, please visit:
  https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices

Or run on an iOS simulator without code signing
════════════════════════════════════════════════════════════════════════════════
Encountered error while building for device.
``` In this instance, the xcresult message is not particularly useful as the more useful messages are displayed at the end.

What do you think about this when parsing the url result:
```
File: file:///path/to/ios/Runner/AppDelegate.swift
StartLineNumber: 11
EndLineNumber: 11
StartColumnNumber: 8
EndColumnNumber: 8
```

I think this detailed compiler error code position is messy enough to deserve a separate PR.

Reply:
2
Good idea, I can create a separate PR to filter errors we don't want to display to user in non-verbose mode.

Reply:
> In this instance, the xcresult message is not particularly useful as the more useful messages are displayed at the end.


Instead of printing as you have, what if instead add the XCResult stuff to the `XcodeBuildResult`?  And then the caller can choose how and in what order to display this.

https://github.com/flutter/flutter/blob/52ae102f182afaa0524d0d01d21b2d86d15a11dc/packages/flutter_tools/lib/src/commands/build_ios.dart#L266

> File: file:///path/to/ios/Runner/AppDelegate.swift
StartLineNumber: 11
EndLineNumber: 11
StartColumnNumber: 8
EndColumnNumber: 8

Let's copy what Xcode does when it displays, with just the line and column start.
`/path/to/ios/Runner/AppDelegate.swift:11:8`

> https://github.com/flutter/flutter/issues/95035

I don't think we should land this PR without making sure confusing `PhaseScriptExecution` errors shows up with dart compilation errors.  We can polish in subsequent PRs, but a dart issue is a very core and much more common case than a Swift compilation error, for example.

Reply:
See `diagnoseXcodeBuildFailure` for where it would probably make sense to handle these from a `XcodeBuildResult`.

Reply:
Also there are a few things in `diagnoseXcodeBuildFailure` that may actually be more accurate once we can parse these xcresults. For example, it just guesses there's a codesigning error sometimes, even when something else really failed.

Reply:
Parsing "url" in this PR: https://github.com/flutter/flutter/pull/95054